### PR TITLE
SDIT-1561 Create DSL for AttributeSearchRequest test data

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/AttributeSearchIntegrationTest.kt [hunk 1]
```diff
@@ -113,34 +79,37 @@ class AttributeSearchIntegrationTest : AbstractSearchDataIntegrationTest() {
...
-      .jsonPath("$.content.length()").isEqualTo(2)
-      .jsonPath("$.content[0].prisonerNumber").isEqualTo("A1234AA")
-      .jsonPath("$.content[1].prisonerNumber").isEqualTo("B1234BB")
+    val request = RequestDsl {
+      query {
+        joinType = OR
+        stringMatcher("firstName" `is` "John")
+        stringMatcher("lastName" `is` "Jones")
+      }
+    }
```
Comment (mikehalmamoj) [inline hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/AttributeSearchIntegrationTest.kt:88]: Use a DSL to generate requests for testing. This is mainly for brevity as some of the longer requests will be huge in JSON - even this simple test has cut from 32 lines of JSON to 6 lines of DSL.

File: hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/AttributeSearchIntegrationTest.kt [hunk 2]
```diff
@@ -113,34 +79,37 @@ class AttributeSearchIntegrationTest : AbstractSearchDataIntegrationTest() {
...
+
+    webTestClient.attributeSearch(request).expectPrisoners("A1234AA", "B1234BB")
```
Comment (mikehalmamoj) [inline hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/AttributeSearchIntegrationTest.kt:90]: All of these search tests will be checking results so added `expectPrisoners` for brevity.

File: hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/AttributeSearchIntegrationTest.kt [hunk 3]
```diff
@@ -1,41 +1,40 @@
...
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.api.AttributeSearchRequest
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.api.JoinType.OR
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.requestdsl.RequestDsl
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.requestdsl.`is`
 
+/**
+ * Test the attribute search engine.
+ *
+ * Note that this class builds AttributeSearchRequest instances using the [RequestDsl]. If you want to know how the requests
+ * look in JSON format see test class [AttributeSearchRequestJsonTest].
```
Comment (mikehalmamoj) [inline hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/AttributeSearchIntegrationTest.kt:22]: Signpost to tests containing actual JSON so we have a few examples.

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**